### PR TITLE
fix: adds ssm from docker image

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
bundles ssm rpm in the docker file to directly copy inside vm 

https://jira.nutanix.com/browse/NCN-101868

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ RUN apk add --no-cache \
     && pip3 install --no-cache-dir --requirement /tmp/requirements.txt \
     && rm -rf /root/.cache
 
+RUN curl -o /opt/amazon-ssm-agent.rpm https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
```
Comment (dkoshkin) [inline Dockerfile:30]: Can we be more consistent with the rest of the steps here please and move this `curl` into the devkit

File: ansible/roles/providers/tasks/aws.yml [hunk 2]
```diff
@@ -42,38 +42,30 @@
...
+    ansible.builtin.yum:
+      name: /opt/amazon-ssm-agent.rpm
+      state: present
+    when: ansible_os_family == "RedHat" or ansible_distribution == "Amazon"
+
+  - name: install aws agents RPM
+    ansible.builtin.zypper:
+      name: /opt/amazon-ssm-agent.rpm
+      state: present
+    when: ansible_os_family == "RedHat" or ansible_distribution == "Amazon"
```
Comment (supershal) [inline ansible/roles/providers/tasks/aws.yml:62]:
```suggestion
    when: ansible_os_family == "Suse" and ansible_distribution != "Amazon"
```

File: ansible/roles/providers/tasks/aws.yml [hunk 3]
```diff
@@ -42,38 +42,30 @@
...
+    copy:
+      src: /opt/amazon-ssm-agent.rpm
+      dest: /opt/amazon-ssm-agent.rpm
 
-- name: install aws agents RPM
+  - name: install aws agents RPM
+    ansible.builtin.yum:
+      name: /opt/amazon-ssm-agent.rpm
+      state: present
+    when: ansible_os_family == "RedHat" or ansible_distribution == "Amazon"
```
Comment (supershal) [inline ansible/roles/providers/tasks/aws.yml:56]:
```suggestion
    when: ansible_os_family == "RedHat" and ansible_distribution != "Amazon"
```

File: ansible/roles/providers/tasks/aws.yml [hunk 4]
```diff
@@ -42,38 +42,30 @@
...
+
+  - name: install aws agents RPM
+    ansible.builtin.zypper:
```
Comment (supershal) [inline ansible/roles/providers/tasks/aws.yml:59]: why we have separate task for Suse? the common `package` module didnt worked for Suse?

Comment (faiq) [reply ansible/roles/providers/tasks/aws.yml:59]: i dont think package supports installing from file.

**Review by supershal — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]

File: Dockerfile.devkit [hunk 5]
```diff
@@ -78,6 +78,7 @@ RUN curl -L "https://github.com/goss-org/goss/releases/download/${GOSS_VERSION}/
 RUN chmod +rx /usr/local/bin/goss-amd64
 ARG BUILDARCH
 RUN ln -s /usr/local/bin/goss-${BUILDARCH} /usr/local/bin/goss
+RUN curl -o /opt/amazon-ssm-agent.rpm https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
```
Comment (dlipovetsky) [inline Dockerfile.devkit:81]: Latest? Don't we want to specify the version?

**Review by msdolbey — APPROVED** [review]
👍 
image is important

**Pull request merged**